tests/functions/test_databricks_client.py

```diff
@@ -0,0 +1,729 @@
+import base64
+import datetime
+import logging
+import uuid
+from contextlib import contextmanager
+from decimal import Decimal
+from importlib.metadata import version
+from typing import Any, Callable, Dict, List, NamedTuple
+from unittest import mock
+
+import pytest
+from databricks.sdk.service.catalog import (
+    ColumnTypeName,
+    CreateFunctionParameterStyle,
+    CreateFunctionRoutineBody,
+    CreateFunctionSecurityType,
+    CreateFunctionSqlDataAccess,
+    DependencyList,
+    FunctionInfo,
+    FunctionParameterInfo,
+    FunctionParameterInfos,
+)
+
+from unitycatalog.functions.databricks import (
+    DEFAULT_EXECUTE_FUNCTION_ARGS,
+    EXECUTE_FUNCTION_ARG_NAME,
+    DatabricksFunctionClient,
+    extract_function_name,
+)
+
+CATALOG = "ml"
+SCHEMA = "serena_uc_test"
+
+_logger = logging.getLogger(__name__)
+
+
+# TODO: CI -- two test cases
+# 1. python 3.10 with databricks-connect 15.1.0, no cluster_id, use serverless
+# 2. python 3.9 with databricks-sdk, with cluster_id
```

Comment:
Even if it's in an early development phase, I'd recommend adding a minimum CI to prevent silly mistakes.

Reply:
Emm it won't work in the CI now, because it needs authentication to a databricks workspace and a warehouse & cluster for testing everything, but I always run the tests locally to make sure they pass before pushing.
<img width="977" alt="image" src="https://github.com/user-attachments/assets/9d760d8b-c18f-46a8-8d98-ac5f8262aad7">

Reply:
Got it. I think you can still test the package installation.

Reply:
> it needs authentication to a databricks workspace and a warehouse

Can we use mocks?

Reply:
We don't need to solve this right now, but imagine an external contributor files a PR in the future. He/she can't access secrets = he/she can't run tests for Databricks.

Reply:
I'll mark those tests involving databricks integration to depend on an environment variable and skip in the normal tests. The rest can be run directly -- mainly for checking the logics.